add torch.chain_matmul to pytorch frontend

4di03 opened the pull request:
Close #20856

ivy-leaves commented:
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

4di03 commented:
HI @ivy-leaves  @YushaArif99,  any updates on this? Would love to get some feedback and start working on some other tasks! Thanks!

YushaArif99 reviewed (approved):
Hi @4di03 

Apologies for the late review. I have been quite swamped with work recently due to which I haven't had the time to review any PRs 😓 
With regards to your PR, the changes lgtm! Could you kindly just address the abovementioned issue. I think we'll then be good to merge 🙂

4di03 commented:
> 

No worries @YushaArif99 ! I added the changes you suggested, let me know if there  are any further issues.

YushaArif99 reviewed (changes requested):
Hi @4di03 

Thanks alot for making the requested changes. I am noticing though that the CI tests for `test_torch_chain_matmul` are failing [here](https://github.com/unifyai/ivy/actions/runs/5859718806/job/15886288306?pr=21013). Its giving a weird error:

```python
E   AssertionError: Frontend function returned non-frontend arrays: ivy.array([[-4., -4.],
E          [-4., -4.]])
```

Can you kindly look into this and re-request a review once these CI tests are passing. Thanks!

4di03 commented:
> Hi @4di03
> 
> Thanks alot for making the requested changes. I am noticing though that the CI tests for `test_torch_chain_matmul` are failing [here](https://github.com/unifyai/ivy/actions/runs/5859718806/job/15886288306?pr=21013). Its giving a weird error:
> 
> ```python
> E   AssertionError: Frontend function returned non-frontend arrays: ivy.array([[-4., -4.],
> E          [-4., -4.]])
> ```
> 
> Can you kindly look into this and re-request a review once these CI tests are passing. Thanks!

Hi @YushaArif99 , thanks for the review. It seems that adding the paddle skip check is causing these tests to fail, so I removed this check and added the changes from https://github.com/unifyai/ivy/pull/21908 which implements multi dot for the paddle backend. All tests seem to be passing locally.

YushaArif99 commented on ivy_tests/array_api_testing/test_array_api:
could you kindly revert these changes to the `test_array_api`

YushaArif99 reviewed (approved):
Hi @4di03 

Thanks a lot for adding the `chain_matmul` implementation to the `paddle` backend. The tests also seem to be passing now. Could you just address the small issue above? After that, PR will be ready to merge.
Thanks! 🙂

4di03 commented:
> #21908

Hi @YushaArif99 , thanks for the review!

It seems that https://github.com/unifyai/ivy/pull/21908 (which implements paddle.multi_dot) was merged, so i restored that paddle implementation as well as test_array_api as you requested. Let me know if this works!

YushaArif99 reviewed (approved):
Thanks a lot for removing the changes to the `test_array_api` files @4di03 
The PR looks good to me. I'll now be going forward with the merge. Great work! 😄
